video/display.h

```diff
@@ -86,20 +91,11 @@ class Display : public Threadable
      */
     bool update()
     {
-        if (!m_Open) {
-            // set opencv window to display full screen
-            cv::namedWindow(WINDOW_NAME, CV_WINDOW_NORMAL);
-            if (m_FullScreen) {
-                setWindowProperty(WINDOW_NAME, cv::WND_PROP_FULLSCREEN, cv::WINDOW_FULLSCREEN);
-            }
-            m_Open = true;
-        }
-
         cv::Mat frame;
         bool newFrame = readFrame(frame);
```

Comment:
Pretty sure this implementation of ``Display::readFrame`` will result in an allocation and copy every frame. Maybe reimplement as:
```c++
virtual bool readFrame(cv::Mat &frame)
{
    // unwrap frame if required
    if (m_Unwrapper && m_ShowUnwrapped) {
        if (!m_VideoInput.readFrame(m_IntermediateFrame)) {
            return false;
        }
        m_Unwrapper->unwrap(m_IntermediateFrame, frame);
    } 
    else {
        if (!m_VideoInput.readFrame(frame)) {
            return false;
        }
    }
    return true;
}
```
Where you always call it with ``readFrame(m_Frame)``

Reply:
Good thinking

Reply:
Actually, on second thoughts, I don't think it will. I just googled it and it seems like if you assign to a matrix it'll only do a shallow copy (which is what I wanted): you have to explicitly call ``copyTo()`` or ``clone()`` if you want to duplicate the data.

Reply:
Good to know :+1: